static/app/components/events/interfaces/spans/spanDetail.tsx

```diff
@@ -434,6 +435,25 @@ class SpanDetail extends React.Component<Props, State> {
     };
   }
 
+  scrollBarIntoView = (spanId: string) => (e: React.MouseEvent<HTMLAnchorElement>) => {
+    // do not use the default anchor behaviour
+    // because it will be hidden behind the minimap
+    e.preventDefault();
+
+    // TODO(txiao): This is causing a rerender of the whole page,
+    // which can be slow.
+    //
+    // make sure to update the location
+    browserHistory.push({
+      ...this.props.location,
+      hash: `#span-${spanId}`,
+    });
+
+    // pushing to browser history does not trigger a hashchange event
+    // so fire one manually
+    window.dispatchEvent(new HashChangeEvent('hashchange'));
```

Comment:
I might be missing something but why are you going through the hash with these listeners? Can't the provider just provide the function which you can call directly with spanId to find the registered scroll function and run it?

If you're trying to handle the anchor auto-scrolling on load, you can check the span id in the url when the provider first receives a span function and call it immediately as soon as it gets passed to register.

Reply:
So there was another place that needed these to be able to scroll to a span but I realize I could just lift the context. This has been fixed in 2864273